.github/workflows/httpclient.yml

```
@@ -22,60 +22,131 @@ jobs:
   rust_check:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - id: run-nats
-      uses: wasmcloud/common-actions/run-nats@main
-    - id: rust-check-action
-      uses: wasmcloud/common-actions/rust-check@main
-      with:
-        working-directory: ${{ env.working-directory }}
+      - uses: actions/checkout@v2
+      - id: run-nats
+        uses: wasmcloud/common-actions/run-nats@main
+      - id: rust-check-action
+        uses: wasmcloud/common-actions/rust-check@main
+        with:
+          working-directory: ${{ env.working-directory }}
+
+  install_cross:
+    if: startswith(github.ref, 'refs/tags/') # Only run on tag push
+    runs-on: ubuntu-latest
+    steps:
+      - uses: wasmcloud/common-actions/install-cross@main
+      - name: Upload cross binary
+        uses: actions/upload-artifact@v3
+        with:
+          name: cross
+          path: ~/.cargo/bin/cross
 
   build_artifact:
-    needs: rust_check
+    needs: install_cross
     if: startswith(github.ref, 'refs/tags/') # Only run on tag push
+    strategy:
+      matrix:
+        target:
+          - x86_64-unknown-linux-gnu
+          - x86_64-apple-darwin
+          - aarch64-unknown-linux-gnu
+          - aarch64-apple-darwin
+          - armv7-unknown-linux-gnueabihf
+          - x86_64-pc-windows-gnu
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download cross binary
+        uses: actions/download-artifact@v3
+        with:
+          name: cross
+          path: ~/.cargo/bin
+
+      - name: Determine artifact name
+        run: |
+          echo "artifact-name=$(cargo metadata --no-deps --format-version 1 | jq -r '.packages[0].targets[0].name')" >> $GITHUB_ENV
+        working-directory: ${{ env.working-directory }}
+
+      - name: Build native executable
+        run: |
+          chmod +x ~/.cargo/bin/cross
+          cross build --release --target ${{ matrix.target }}
+        working-directory: ${{ env.working-directory }}
+
+      - name: Upload executable to GH Actions
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.target }}
+          if-no-files-found: error
+          path: |
+            ${{ env.working-directory }}/target/${{ matrix.target }}/release/${{ env.artifact-name }}
+            ${{ env.working-directory }}/target/${{ matrix.target }}/release/${{ env.artifact-name }}.exe
```

Could this be simplified to one line with a wildcard, or are there other files that would match?

```suggestion
            ${{ env.working-directory }}/target/${{ matrix.target }}/release/${{ env.artifact-name }}*
```